### PR TITLE
Add the initial backend wiring and navigation

### PR DESCRIPTION
**Pull request by jpellizzari**
Adds an RPC codegen'd transport layer using a library I have had good results with in the past called [twirp](https://github.com/twitchtv/twirp). This RPC framework supports HTTP 1.1, which means it work in the browser (unlike `grpc`). This will hopefully cut down on mismatched types and require less AJAX boilerplate (see first 4 commits).

If this approach is agreeable, I will add a dockerized workflow to ensure contributors don't have to install a lot of `protoc` dependencies required to run the `make proto-native` job.

This PR also adds the initial navigation pattern with placeholder pages:
Screenshot from 2020-12-08 13-11-51

### CODE REVIEW

**Review by bia — APPROVED** [review]
🚀

File: README.md [hunk 1]
```diff
@@ -10,3 +10,17 @@ To set up a development environment
...
+go get -u github.com/golang/protobuf/protoc-gen-go
+go get -u github.com/twitchtv/twirp/protoc-gen-twirp
+go get -u go.larrymyers.com/protoc-gen-twirp_typescript
+```
+
+Then run:
+
+```shell
+make proto-native
+```
```
Comment (bia) [inline README.md:26]: Might need to add another `make dev` here.

Comment (jpellizzari) [reply README.md:26]: @bia The generated protocol files are already checked into version control, so no need to generate them unless there are changes to the .proto files.

File: .gitignore [hunk 2]
```diff
@@ -1,5 +1,4 @@
 node_modules
 .cache
 dist
-bin
-static
+bin
```
Comment (scottrigby) [inline .gitignore:4]: Need newline at EOF

File: pkg/rpc/clusters/clusters.proto [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    string name = 1;
+}
+
+message ListContextsReq {
+
+}
+
+message ListContextsRes {
+    repeated Context contexts = 1;
+}
```
Comment (scottrigby) [inline pkg/rpc/clusters/clusters.proto:19]: Needs newline at EOF

File: ui/lib/rpc/clusters.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+            if (!resp.ok) {
+                return throwTwirpError(resp);
+            }
+
+            return resp.json().then(JSONToListContextsRes);
+        });
+    }
+    
+}
+
```
Comment (scottrigby) [inline ui/lib/rpc/clusters.ts:90]: Unneeded extra newline

File: ui/lib/util.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+export const wrappedFetch = (url, opts: RequestInit = {}) => {
+    return fetch(url, {
+      ...opts,
+      credentials: "same-origin",
+      headers: {
+        "content-type": "application/json",
+        ...(opts.headers || {}),
+      },
+    });
+  };
```
Comment (scottrigby) [inline ui/lib/util.ts:10]: Needs newline at EOF

**Review by scottrigby — COMMENTED** [review]
Just a few minor EOF issues

**Comment by scottrigby** [comment]
One other thought, you need the gitignored dist dir. You could either add that kept dir to git with just ignored contents, or `make` as needed like

```diff
diff --git a/Makefile b/Makefile
index 91d93ac..1bdeac3 100644
--- a/Makefile
+++ b/Makefile
@@ -7,6 +7,7 @@ test:
        go test ./..
 
 assets: ui
+       mkdir -p dist
        go run -tags=generate pkg/assets/generate.go
 
 build: assets
```

File: ui/components/LeftNav.tsx [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+            {_.map(contexts, (c) => (
+              <MenuItem value={c.name} key={c.name}>
+                {c.name}
+              </MenuItem>
+            ))}
+          </Select>
+        </FormControl>
+      </div>
+      <div>
+        <Tabs centered={false} orientation="vertical" value={normalizedPath}>
```
Comment (scottrigby) [inline ui/components/LeftNav.tsx:72]: I know this is preliminary, so just noting that the context select above resets when clicking each tab placeholder

Comment (jpellizzari) [reply ui/components/LeftNav.tsx:72]: Yep, working as intended for now. The next PR will address UI nav.

**Comment by jpellizzari** [comment]
> One other thought, you need the gitignored dist dir. You could either add that kept dir to git with just ignored contents, or `make` as needed like
> 
> ```diff
> diff --git a/Makefile b/Makefile
> index 91d93ac..1bdeac3 100644
> --- a/Makefile
> +++ b/Makefile
> @@ -7,6 +7,7 @@ test:
>         go test ./..
>  
>  assets: ui
> +       mkdir -p dist
>         go run -tags=generate pkg/assets/generate.go
>  
>  build: assets
> ```

The `make ui` job will create the `dist` directory. There may be some makefile magic that needs to be added to make it seamless.

**Review by stefanprodan — APPROVED** [review]
LGTM

**Pull request merged**